app/web/features/messages/requests/HostRequestSendField.tsx

```diff
@@ -175,99 +188,85 @@ export default function HostRequestSendField({
-                onConfirm={handleReject}
-              >
-                {(setIsOpen) => (
-                  <FieldButton
-                    isLoading={isButtonLoading}
-                    callback={() => setIsOpen(true)}
-                  >
-                    {t("messages:close_request_button_text")}
-                  </FieldButton>
-                )}
-              </ConfirmationDialogWrapper>
-            )}
-            {isReferenceAvailable && (
-              <Link
-                href={{
-                  pathname: `${leaveReferenceBaseRoute}/${
-                    referenceTypeRoute[ReferenceType.REFERENCE_TYPE_HOSTED]
-                  }/${hostRequest.surferUserId}/${hostRequest.hostRequestId}`,
-                }}
-                passHref
-              >
-                <StyledButton color="primary" component="a">
-                  {t("messages:write_reference_button_text")}
-                </StyledButton>
-              </Link>
-            )}
-          </>
-        ) : (
-          //user is the surfer
-          <>
-            {hostRequest.status ===
-              HostRequestStatus.HOST_REQUEST_STATUS_ACCEPTED && (
-              <FieldButton callback={handleConfirm} isLoading={isButtonLoading}>
-                {t("messages:confirm_request_button_text")}
-              </FieldButton>
-            )}
-            {(hostRequest.status ===
-              HostRequestStatus.HOST_REQUEST_STATUS_PENDING ||
-              hostRequest.status ===
-                HostRequestStatus.HOST_REQUEST_STATUS_ACCEPTED ||
-              hostRequest.status ===
-                HostRequestStatus.HOST_REQUEST_STATUS_REJECTED ||
-              hostRequest.status ===
-                HostRequestStatus.HOST_REQUEST_STATUS_CONFIRMED) && (
-              <ConfirmationDialogWrapper
-                title={t("messages:close_request_dialog_title")}
-                message={t("messages:close_request_dialog_surfer")}
-                onConfirm={handleCancel}
-              >
-                {(setIsOpen) => (
-                  <FieldButton
-                    isLoading={isButtonLoading}
-                    callback={() => setIsOpen(true)}
-                  >
-                    {t("global:cancel")}
-                  </FieldButton>
-                )}
-              </ConfirmationDialogWrapper>
-            )}
-            {isReferenceAvailable && (
-              <Link
-                href={{
-                  pathname: `${leaveReferenceBaseRoute}/${
-                    referenceTypeRoute[ReferenceType.REFERENCE_TYPE_SURFED]
-                  }/${hostRequest.hostUserId}/${hostRequest.hostRequestId}`,
-                }}
-                passHref
-              >
-                <StyledButton color="primary" component="a">
-                  {t("messages:write_reference_button_text")}
-                </StyledButton>
-              </Link>
-            )}
-          </>
+        {!isPast &&
+          (isHost ? (
```

Comment:
Can you turn `!isPast && isHost` into a variable that's like `const showResponseButton` (or whatever it is) and use that instead? Just to make it a bit cleaner as it's already an unruly component.

Reply:
The `!isPast` applies to both the `isHost` and `!isHost` below... I could move them into another function/component in this file though to make it less crazy. Agree it's unwieldly.